torch/_inductor/async_compile.py

```diff
@@ -174,20 +178,45 @@ def save(kernel_src: str, future: LambdaFuture):
         CompiledTritonKernels._cache[key] = future
 
     @staticmethod
-    def get(kernel_src: str, default: Any) -> LambdaFuture:
+    def get(kernel_src: str) -> Optional[CacheResult]:
         key = CompiledTritonKernels.key(kernel_src)
-        return CompiledTritonKernels._cache.get(key, default)
+        result = CompiledTritonKernels._cache.get(key, None)
+        if result is not None and isinstance(result, CachingAutotuner):
+            # Return a copy so we don't pollute the cached result
+            return copy.copy(result)
+        return result
 
     @staticmethod
     def cache_clear():
         CompiledTritonKernels._cache = {}
 
     @staticmethod
-    def remove_future(kernel_src: str) -> None:
+    def remove_future(
+        kernel_src: str, compiled_kernel: Optional[CachingAutotuner]
+    ) -> None:
         key = CompiledTritonKernels.key(kernel_src)
-        if key in CompiledTritonKernels._cache:
+
+        # Delete the LambdaFuture if there is one
+        if key in CompiledTritonKernels._cache and isinstance(
+            CompiledTritonKernels._cache[key], LambdaFuture
+        ):
             del CompiledTritonKernels._cache[key]
 
+        # Replace it with the fully compiled kernel if one exists
+        if compiled_kernel is not None and compiled_kernel.is_statically_launchable():
```

Comment:
This logic is a little gross and idk if I like it: basically, after I get the compiled triton kernels back from async_compile, we store it in the cache so FXGraphCache can save it later. Then on FXGraphCache load, we put these compiled kernels into the the cache. 

This allows pretty seamless integration with FXGraphCache, but at the cost of having to make CompiledTritonKernels take in both Futures and non Futures, which is confusing. 

We could instead have the cache only store LambdaFutures, and make dummy LambdaFutures to unwrap and wrap from the cache? But that has its own icky-ness I don't like. Feedback appreciated